src/Mvc/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets

```diff
@@ -0,0 +1,39 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <PropertyGroup>
+    <_OpenApiDocumentsCache>$(BaseIntermediateOutputPath)$(MSBuildProjectName).OpenApiFiles.cache</_OpenApiDocumentsCache>
+  </PropertyGroup>
+  <ItemGroup Condition=" '$(OpenApiGenerateDocuments)' == 'true' ">
+    <ProjectCapability Include="OpenApiGenerateDocuments" />
+  </ItemGroup>
+
+  <Target Name="OpenApiGetDocuments" Returns="@(_OpenApiProjectDocuments)">
+    <ReadLinesFromFile File="$(_OpenApiDocumentsCache)">
+      <Output TaskParameter="Lines" ItemName="_OpenApiProjectDocuments" />
+    </ReadLinesFromFile>
+  </Target>
+
+  <Target Name="GenerateOpenApiDocuments" Condition=" '$(OpenApiGenerateDocuments)' == 'true' ">
+    <PropertyGroup>
+      <_Command>dotnet "$(MSBuildThisFileDirectory)/../tools/dotnet-getdocument.dll" --assembly "$(TargetPath)"</_Command>
+      <_Command>$(_Command) --file-list "$(_OpenApiDocumentsCache)" --framework "$(TargetFrameworkMoniker)"</_Command>
+      <_Command>$(_Command) --output "$(OpenApiDocumentsDirectory)" --project "$(MSBuildProjectName)"</_Command>
+      <_Command Condition=" '$(ProjectAssetsFile)' != '' ">$(_Command) --assets-file "$(ProjectAssetsFile)"</_Command>
+      <_Command Condition=" '$(PlatformTarget)' != '' ">$(_Command) --platform "$(PlatformTarget)"</_Command>
+      <_Command Condition=" '$(PlatformTarget)' == '' AND '$(Platform)' != '' ">$(_Command) --platform "$(Platform)"</_Command>
+      <_Command Condition=" '$(RuntimeIdentifier)' != '' ">$(_Command) --runtime "$(RuntimeIdentifier)"</_Command>
+      <_Command>$(_Command) $(OpenApiGenerateDocumentsOptions)</_Command>
+    </PropertyGroup>
+
+    <Message Importance="high" Text="%0ARetrieveOpenApiDocuments:" />
+    <Message Importance="high" Text="  $(_Command)" />
+    <Exec Command="$(_Command)" />
```

Comment:
What are the caching expectations of this? Should we be adding some things to `FileWrites` so they can be removed by incremental clean?

Reply:
I have been thinking about `FileWrites` both here and in the client-side glue. But, every time I start making those changes, I hit the "What if the user plans to check these files in?" question. These files aren't always throw-away. But, I could probably complicate things enough to avoid that concern e.g. with conditions on `$(OpenApiDocumentsDirectory)` matching the default.

I'll add a note to #4923 reminding me to work through the implications and do _something_.

Reply:
Discussed in person. Right now there is no input/output checking and this command will re-run every time you build, even if the assembly does not change.

This is tracked separately and will be improved in the next preview.